src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -120,6 +120,7 @@
    :producer_timestamp pls/Timestamp
    :producer (s/maybe s/Str)
    :code_id (s/maybe s/Str)
+   :job_id (s/maybe s/Str)
```

Comment:
Can we make this an optional-key? Seems better to keep backward compatibility where possible.